app/services/task_service/gpt_generate_tests.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module TaskService
+  class GptGenerateTests < ServiceBase
+    def initialize(task:)
+      super()
+      @task = task
+      @message = task.description
+      @language = @task.programming_language.language || 'Python'
+    end
+
+    def execute
+      file_content = gpt_response
+      test_file = TaskFile.new(content: file_content, name: 'test.py', used_by_grader: true, visible: false, xml_id: SecureRandom.uuid)
+      testing_framework = TestingFramework.find_or_initialize_by(name: 'Pytest', version: 6)
```

Comment:
Similar to above, the test framework might not be suitable.

Reply:
see above comment :)

Reply:
Thanks, I would go with a database entry, too.

Reply:
The testing framework is no longer set at all, and since it is optional, we can resolve this thread.